eos_data_distribution/ndn/file.py

```diff
@@ -274,6 +264,44 @@ def _on_complete(self):
         os.unlink(self._sgt_filename)
         super(FileConsumer, self)._on_complete()
 
+    def _create_files(self, filename):
+        self._filename = filename
+        mkdir_p(os.path.dirname(filename))
+        self._part_filename = '%s.part' % (filename, )
+        self._part_fd = os.open(
+            self._part_filename, os.O_CREAT | os.O_WRONLY | os.O_NONBLOCK, 0o600)
+        self._sgt_filename = '%s.sgt' % (filename, )
+        self._sgt_fd = os.open(
+            self._sgt_filename, os.O_CREAT | os.O_RDWR, 0o600)
+
+
+class FileConsumer(Consumer):
+
+    """Write to a File"""
+
+    def __init__(self, name, filename, *args, **kwargs):
+        self._create_files(filename)
+
+        super(FileConsumer, self).__init__(name, *args, **kwargs)
+
+
+class DirConsumer(Consumer):
+
+    """Write to a Directory routing the name"""
+
+    def __init__(self, name, dirname, *args, **kwargs):
+        mkdir_p(dirname)
+        self._dirname = dirname
+        super(DirConsumer, self).__init__(name, *args, **kwargs)
+
+    def _on_data(o, interest, data):
+        self._create_files(
+            path.join(self._dirname, chunks.get_chunkless_name(
+                interest.getName())))
+
+        super(DirConsumer, self)._on_data(o, interest, data)
+        self._on_data = super(DirConsumer, self)._on_data
```

Comment:
we don't, this is a consumer, we should not recieve different Data (can't recieve interests) packets with different names.
that is because once we get the first interest with a full name (i.e. timestamped) *we* ask for new interests on that name.